reconfigure.bash

```diff
@@ -2,6 +2,13 @@
 
 set -o errexit
 
+if [[ $# -ne 1 ]] || [[ $1 != "master" && $1 != "agent" && $1 != "repo" ]];
```

Comment:
The test `[[ $# -ne 1 ]]` prevents the script from being run with no arguments when a role is already saved to the `role` file.

If possible I would like to preserve the ability to reconfigure without specifying an argument as it means we only need one common version of the reconfigure command line rather than one per role.

https://github.com/ros-infrastructure/buildfarm_deployment_config/blob/bf27c21108e74b1aa57ee4d6b9d8a6e7dc15fd05/hiera/hieradata/common.yaml#L28